ols/app/endpoints/ols.py

```diff
@@ -142,6 +143,20 @@ def store_conversation_history(
 
 def validate_question(conversation_id: str, llm_request: LLMRequest) -> str:
     """Validate user question, raise HTTPException in case of any problem."""
+    # Redact the input question
+    try:
+        redacted_query = QuestionFilter.setup().redact_question(llm_request.query)
```

Comment:
i don't think this should be called from within validate_question, validating questions and filtering/redacting questions should be considered peer steps in the query flow.  This current implementation couples them together in an undesirable way

Reply:
also the fact that setup() is being called on every request is not ideal, we should only need to call it once during OLS initialization.

Reply:
@bparees I was expecting this comment. When I looked it , we need to make use of fastapi lifespan to do this . https://fastapi.tiangolo.com/advanced/events/#async-context-manager . I would like this to be separate from this PR . as I feel that , I will touch a lot of files.

Reply:
doesn't seem like we have to do it that way.  We load/init our config only a single time and then share/reuse it.  Even the docs you linked to say:

```
You could load it at the top level of the module/file, but that would also mean that it would load the model even if you are just running a simple automated test, then that test would be slow because it would have to wait for the model to load before being able to run an independent part of the code.
```

the drawbacks don't seem to me to apply to this init logic, nor do we care about cleaning it up on shutdown, so the lifecycle events seem like a nice way to handle this, but certainly not the only way and i'm not convinced it justifies a refactor that would touch a lot of files.


but i am glad to hear you anticipated the comment and had thought about how to address it!

Reply:
Loaded part of init function.